components/export/api/src/main/java/org/phenotips/export/internal/DataToCellConverter.java

```diff
@@ -1215,6 +1215,85 @@ public DataSection isNormalBody(Patient patient)
         return bodySection;
     }
 
+
+    public DataSection isSolvedHeader(Set<String> enabledFields) throws Exception
+    {
+        String sectionName = "isSolved";
+
+        // Must be linked to keep order; in other sections as well
+        Map<String, String> fieldToHeaderMap = new LinkedHashMap<>();
+        fieldToHeaderMap.put("solved", "Solved");
+        fieldToHeaderMap.put("solved__pubmed_id", "PubMed ID");
+        fieldToHeaderMap.put("solved__gene_id", "Gene ID");
+        fieldToHeaderMap.put("solved__notes", "Notes");
+
+        Set<String> present = new LinkedHashSet<>();
+        for (String fieldId : fieldToHeaderMap.keySet()) {
+            if (enabledFields.remove(fieldId)) {
+                present.add(fieldId);
+            }
+        }
+        this.enabledHeaderIdsBySection.put(sectionName, present);
+
+        DataSection headerSection = new DataSection();
+        if (present.isEmpty()) {
+            return null;
+        }
+
+        int x = 0;
+        for (String fieldId : present) {
+            DataCell headerCell = new DataCell(fieldToHeaderMap.get(fieldId), x, 1, StyleOption.HEADER);
+            headerSection.addCell(headerCell);
+            x++;
+        }
+        DataCell headerCell = new DataCell("Solved Status", 0, 0, StyleOption.LARGE_HEADER);
+        headerCell.addStyle(StyleOption.HEADER);
+        headerSection.addCell(headerCell);
+
+        return headerSection;
+    }
+
+    public DataSection isSolvedBody(Patient patient)
+    {
+        String sectionName = "isSolved";
+        Set<String> present = this.enabledHeaderIdsBySection.get(sectionName);
+        if (present == null || present.isEmpty()) {
+            return null;
+        }
+        DataSection bodySection = new DataSection();
+        PatientData<String> patientData = patient.getData("solved");
+
+        Integer x = 0;
+        if (present.contains("solved")) {
```

Comment:
These are left uncolored on purpose.
I'm assuming that majority of cases will be unsolved, and it could be psychologically draining to see too much red.